lib/config_read.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+$LOAD_PATH.push('fluentd/lib')
+
+require 'fluent/config/v1_parser'
+require 'optparse'
+
+# Accepts a file path and prints out parsed version
+class ConfigFile
+  def initialize(argv = ARGV)
+    @argv = argv
+    prepare_option_parser
+    parse_options
+    print_configurations(@file_parse)
+  end
+
+  def prepare_option_parser
+    @parser = OptionParser.new
+    @parser.banner = "\nPrint-Config-File Tool\nUsage: #{$PROGRAM_NAME} " \
+      "path/to/file\nOutput: Parsed version of config file"
+    @parser.parse!(@argv)
+  end
+
+  def usage(message = nil)
+    puts @parser.to_s
+    puts "\nError: #{message}" if message
+    exit(false)
+  end
+
+  def parse_options
+    # parses the arguments, quits program if arguments are invalid
+    raise 'Must specify path of file' if @argv.empty?
+    raise 'Only one argument is needed' if @argv.size > 1
+    raise 'Enter a valid file path' unless File.exist?(@argv[0]) &&
+                                           (File.extname(@argv[0]) == '.conf')
+
+    # extracts required information and parses the config file
```

Comment:
`file_str`, `file_name` and `file_dir` are variables only used in this function, so they don't need to be instance variable (a.k.a. `@xxx`)

Reply:
Sounds good, will change that.